Documentation/CRDs/Cluster/external-cluster/external-cluster.md

```diff
@@ -305,3 +286,63 @@ you can export the settings from this cluster with the following steps.
+
+    ```console
+    python3 create-external-cluster-resources.py --upgrade
+    ```
+
+2. If the consumer cluster has restricted caps
+
+    Restricted users created using `--restricted-auth-permission` flag need to pass mandatory flags: '`--rbd-data-pool-name`(if it is a rbd user), `--k8s-cluster-name` and `--run-as-user`' flags while upgrading, in case of cephfs users if you have passed `--cephfs-filesystem-name` flag while creating csi-users then while upgrading it will be mandatory too. In this example the user would be `client.csi-rbd-node-rookstorage-replicapool` (following the pattern `csi-user-clusterName-poolName`)
+
+    ```console
+    python3 create-external-cluster-resources.py --upgrade --rbd-data-pool-name replicapool --k8s-cluster-name rookstorage --run-as-user client.csi-rbd-node-rookstorage-replicapool
+    ```
+
+    !!! note
+        An existing non-restricted user cannot be converted to a restricted user by upgrading.
+        The upgrade flag should only be used to append new permissions to users. It shouldn't be used for changing a csi user already applied permissions. For example, you shouldn't change the pool(s) a user has access to.
+
+
+### Upgrade cluster to utilize new feature
+
+Some Rook upgrades may require re-running the import steps, or may introduce new external cluster features that can be most easily enabled by re-running the import steps.
+
+To re-run the import steps with new options, the python script should be re-run using the same configuration options that were used for past invocations, plus the configurations that are being added or modified.
+
+Starting with Rook v1.15, the script stores the configuration in the external-cluster-user-command configmap for easy future reference.
+
+* arg: Exact arguments that were used for for processing the script. Argument that are decided using the Priority: command-line-args > config.ini file values > default values.
+
+* command: The exact python command with the command arguments.
+
+Example `external-cluster-user-command` ConfigMap:
+
+```console
+$ kubectl get cm  -nrook-ceph external-cluster-user-command  -o yaml
+apiVersion: v1
+data:
+  arg: '[''verbose: False'', ''rgw-pool-prefix: default'', ''restricted-auth-permission:
+    False'', ''v2-port-enable: False'', ''format: bash'', ''cephfs-filesystem-name:
+    ocs-storagecluster-cephfilesystem'', ''cephfs-metadata-pool-name: ocs-storagecluster-cephfilesystem-metadata'',
+    ''cephfs-data-pool-name: ocs-storagecluster-cephfilesystem-data0'', ''rbd-data-pool-name:
+    ocs-storagecluster-cephblockpool'', ''rgw-skip-tls: False'', ''skip-monitoring-endpoint:
+    False'', ''dry-run: False'', ''upgrade: False'']'
+  command: --rbd-data-pool-name ocs-storagecluster-cephblockpool --format bash
+kind: ConfigMap
+metadata:
+  creationTimestamp: "2024-07-03T14:39:52Z"
+  name: external-cluster-user-command
+  namespace: rook-ceph
+  resourceVersion: "478493"
+  uid: d9643038-9850-4f1f-8974-17ca6571b93e
+```
```

Comment:
Instead of this, can we just show a command users can run to get the `.command` ?

Reply:
From now , The python command can have the --config-file set and which user might not remember,

So showing what exact arg make sense

Reply:
So they maybe create a whole new config from that values and don't bother of the command args separately

Reply:
There's a lot of text here that I don't think makes sense for users to see. Most of the metadata, and the `arg` don't really apply I don't think. For the user to make use of `arg` they need to do special processing, which is non-ideal. I'm not sure why `arg` is even in the configmap in this form, other than to be informational.

Reply:
The data of configmap is always the string, `data field is designed to contain UTF-8 strings`

`rendered and usable config file` -> is there a way for this?? Or you suggest to do it by string formatting?

Reply:
The `arg` is given as what appears to be a json blob -- probably just python output of a list I would guess. This isn't usable for users without them having to do string manipulation. We should render it in INI format for it to be useful to them.

Reply:
Adding clarity:

From my perspective, users should be able to copy the config file from the configmap directly, without doing any of their own text manipulation. If they have to modify the text to put it into a config file, I think most users simply won't do it, and if they have used the `--config` arg, then as Parth mentioned information would be lost.

My ideal would be that users simply copy the INI-formatted config from the configmap and then paste it into a file to use in the `--config` arg the next time they run the script. They would also have the opportunity to update/edit configs in the file, or to supply new/additional CLI flags at this time.

If it becomes too complicated for us to render the INI-formatted config into the configmap, then I would propose we drop config file functionality to avoid the info loss issue Parth mentioned. I really do not believe users will be inclined to use the `arg` field as I see it here, and I want to avoid causing users confusion that might lead to errors.

Reply:
updated and add a pr for arg https://github.com/rook/rook/pull/14464

Reply:
Done